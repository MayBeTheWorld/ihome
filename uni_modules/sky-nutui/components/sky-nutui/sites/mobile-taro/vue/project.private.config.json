{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}