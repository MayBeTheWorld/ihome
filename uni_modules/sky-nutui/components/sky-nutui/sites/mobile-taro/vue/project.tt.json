{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
