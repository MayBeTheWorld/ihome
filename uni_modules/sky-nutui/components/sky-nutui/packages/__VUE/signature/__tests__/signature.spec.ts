import { mount, config } from '@vue/test-utils';
import Signature from '../index.vue';
import NutIcon from '../../icon/index.vue';
import NutButton from '../../button/index.vue';
import { nextTick, reactive, toRefs } from 'vue';
import { createCanvas } from 'canvas';

function sleep(delay = 0): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}
// 所有的测试用例之前执行一次
beforeAll(() => {
  config.global.components = {
    NutSignature: Signature,
    NutIcon,
    NutButton
  };
});
// 所有的测试用例之后执行一次
afterAll(() => {
  config.global.components = {};
});

test('props custom-class', async () => {
  const component = {
    template: `<nut-signature :custom-class="customClass" :line-width="lineWidth" :stroke-style="strokeStyle"></nut-signature>`,
    setup() {
      const data = reactive({
        customClass: 'test',
        lineWidth: 5,
        strokeStyle: '#333'
      });
      return {
        ...toRefs(data)
      };
    }
  };
  const wrapper = mount(component);
  await nextTick();
  const signatureWrapper = wrapper.findAll('.nut-signature');
  expect(signatureWrapper[0].classes()).toContain(wrapper.vm.customClass);

  const canvas = createCanvas(200, 200);
  const ctx = canvas.getContext('2d');
  ctx.lineWidth = wrapper.vm.lineWidth;
  ctx.strokeStyle = wrapper.vm.strokeStyle;
  ctx.beginPath();
  ctx.lineTo(10, 50);
  ctx.lineTo(80, 120);
  ctx.stroke();
  const _img = document.createElement('img');
  _img.src = canvas.toDataURL();
  expect(canvas.toDataURL()).not.toBeNull();
  expect(wrapper.html()).toMatchSnapshot();
});
