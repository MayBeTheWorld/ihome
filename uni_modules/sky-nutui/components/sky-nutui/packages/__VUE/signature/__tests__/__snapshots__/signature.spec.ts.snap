// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props custom-class 1`] = `
"<div class=\\"nut-signature test\\">
  <div class=\\"nut-signature-inner\\"><canvas height=\\"0\\" width=\\"0\\"></canvas></div>
  <view class=\\"nut-button nut-button--default nut-button--normal nut-button--round nut-signature-btn\\">
    <view class=\\"nut-button__warp\\">
      <!--v-if-->
      <!--v-if-->
      <view class=\\"\\">重签</view>
    </view>
  </view>
  <view class=\\"nut-button nut-button--primary nut-button--normal nut-button--round nut-signature-btn\\">
    <view class=\\"nut-button__warp\\">
      <!--v-if-->
      <!--v-if-->
      <view class=\\"\\">确认</view>
    </view>
  </view>
</div>"
`;
